### PR TITLE
feat: Web request checkpoint deletion [DET-7113]

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Support request checkpoint deletion on web

## Test Plan

Choose experiments with checkpoints and request checkpoint deletion. 
Screen Shot 2022-07-14 at 1 03 43 PM

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/useModal/Checkpoint/useModalCheckpoint.tsx [hunk 1]
```diff
@@ -83,6 +85,32 @@ const useModalCheckpoint = ({
...
   const handleOk = useCallback(() => onClose?.(ModalCloseReason.Ok), [ onClose ]);
 
+  const handleDelete = useCallback(() => {
+    if(!checkpoint.uuid) return;
+    readStream(detApi.CheckPoint.deleteCheckpoints({ checkpointUuids: [ checkpoint.uuid ] }));
+  }, [ checkpoint ]);
+
+  const deleteCPModalProps: ModalFuncProps = useMemo(() => {
+    const content =
+      `Are you sure you want to request checkpoint deletetion for batches
```
Comment (mapmeld) [inline webui/react/src/hooks/useModal/Checkpoint/useModalCheckpoint.tsx:95]: nitpick: should be deletion

File: webui/react/src/hooks/useModal/Checkpoint/useModalCheckpoint.module.scss [hunk 2]
```diff
@@ -6,6 +6,17 @@
 
   .row {
     display: flex;
+
+    .size {
+      align-items: center;
+      display: flex;
+
+      :global {
```
Comment (mapmeld) [inline webui/react/src/hooks/useModal/Checkpoint/useModalCheckpoint.module.scss:14]: I think that this :global section is not needed?

Comment (mapmeld) [reply webui/react/src/hooks/useModal/Checkpoint/useModalCheckpoint.module.scss:14]: oh, I thought it was the footer buttons, now I see that it is for the "Request Checkpoint Deletion" button, my bad

File: webui/react/src/services/apiConfig.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ const generateApiConfig = (apiConfig?: Api.ConfigurationParameters) => {
   const config = updatedApiConfigParams(apiConfig);
   return {
     Auth: new Api.AuthenticationApi(config),
+    CheckPoint: Api.CheckpointsApiFetchParamCreator(config),
```
Comment (mapmeld) [inline webui/react/src/services/apiConfig.ts:27]: my opinion is for consistency Checkpoint should not have uppercase P (here or in `useModalCheckpoint.tsx`)

File: webui/react/src/hooks/useModal/Checkpoint/useModalCheckpoint.tsx (hunk 1 above)
Comment (gt2345) [reply webui/react/src/hooks/useModal/Checkpoint/useModalCheckpoint.tsx:95]: 😂 fixed..

**Review by mapmeld — APPROVED** [review]

**Pull request merged**